compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -131,3 +132,41 @@ private val truePredicate: (Any?) -> Boolean = { true }
  * method throws an exception if more than one element is found.
  */
 internal fun <T> Iterable<T>.singleOrEmpty(): T? = singleOrEmpty(truePredicate)
+
+internal fun TypeName.toStringOfSimpleNamesOnly(): String {
+  return removePackageNames(toString())
+}
+
+internal fun removePackageNames(type: String): String {
+  if (type.isEmpty()) return ""
+
+  val builder = StringBuilder()
+  var current = ""
+  for (i in type.indices) {
+    if (i == type.lastIndex) {
+      builder.append(current + type[i])
+      break
+    }
+    when (val char = type[i]) {
+      '.' -> {
+        current = ""
+      }
+
+      ',', '<', '>' -> {
+        builder.append(current + char)
+        current = ""
+      }
+
+      ' ' -> {
+        builder.append(char)
+        current = ""
+      }
+
+      else -> {
+        current += char
+      }
+    }
+  }
+
+  return builder.toString()
```

Comment:
I think you could simplify this using a regex, something like `type.replace("([a-z]+\\.)+".toRegex(), "")`. Probably needs a little more tinkering to handle nested class cases like com.squareup.Foo.Bar

Reply:
Removed this completely